easybuild/framework/easyconfig/format/yeb.py

```diff
@@ -94,6 +95,10 @@ def parse(self, txt):
         """
         txt = self._inject_constants_dict(txt)
         self.parsed_yeb = yaml.load(txt)
+        # parse dependencies upfront; this can't be done with simple type checking as it is already a list,
+        # but needs to be a list of tuples instead of dicts.
```

Comment:
wasn't the actual reason more like making sure that the conversion happens early enough?

we can detect a list of dicts and acts accordingly in the type conversion stuff, that's not the real problem?